sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/moduls/PrepareWrapperGIT.java

```diff
@@ -0,0 +1,200 @@
+        String repositoryURL = gitContext.location;
+        List<String> commands = new ArrayList<>();
+        if (gitContext.credentialMap != null && !gitContext.credentialMap.isEmpty()) {
+            repositoryURL = getRepositoryURL(gitContext.location);
+        }
+
+        commands.add("bash");
+        commands.add("-c");
+        if (gitContext.cloneWithoutHistory) {
+            commands.add("git clone --depth 1 " + repositoryURL + " " + gitContext.pdsPrepareUploadFolderDirectory);
+        } else {
+            commands.add("git clone " + repositoryURL + " " + gitContext.pdsPrepareUploadFolderDirectory);
+        }
+
+        ProcessBuilder builder = new ProcessBuilder(commands);
+
+        Map<String, String> environment = builder.environment();
+        if (gitContext.credentialMap != null && !gitContext.credentialMap.isEmpty()) {
+            for (Map.Entry<String, SealedObject> entry : gitContext.credentialMap.entrySet()) {
+                try {
+                    environment.put(entry.getKey(), CryptoAccess.CRYPTO_STRING.unseal(entry.getValue()));
+                } catch (Exception e) {
+                    LOG.error("Error while unsealing credential: " + entry.getKey(), e);
+                    throw new IOException("Error while unsealing credential: " + entry.getKey(), e);
+                }
+            }
+        }
+
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (IOException e) {
+            LOG.error("Error while cloning repository: " + gitContext.location, e);
+            throw new IOException("Error while cloning repository: " + gitContext.location + "GIT Error: " + e.getMessage(), e);
+        }
+        waitForProcessToFinish();
+    }
+
+    public void cleanGitDirectory(String pdsPrepareUploadFolderDirectory) throws IOException {
+        List<String> commands = new ArrayList<>();
+        // removes recursively all git files from a directory
+        // ( find . -type d -name ".git" && find . -name ".gitignore" && find . -name
+        // ".gitmodules" ) | xargs rm -rf
+
+        File uploadDir;
+        uploadDir = Paths.get(pdsPrepareUploadFolderDirectory).toAbsolutePath().toFile();
+
+        commands.add("bash");
+        commands.add("-c");
+        commands.add("( find . -type d -name .git && find . -name .gitignore && find . -name .gitattributes ) | xargs rm -rf");
+
+        ProcessBuilder builder = new ProcessBuilder(commands);
+        builder.directory(uploadDir);
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (Exception e) {
+            LOG.error("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+            throw new IOException("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+        }
+        waitForProcessToFinish();
+    }
+
+    String getRepositoryURL(String location) {
```

Comment:
This logic could be inside its own class `SafeAuthenticatedRepositoryURLFactory` and only be used here.

Try to separate logic - Makes parts smaller and easier to refactor and to maintain.

e.g. (without injection and static, but the idea becomes clear...)
```java

//....
String authenticatedLocation = SafeAuthenticatedRepositoryURLFactory.create(location);
```